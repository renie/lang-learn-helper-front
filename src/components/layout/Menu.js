import React from 'react'

import MenuItem from './MenuItem'
import menuItems from './menuItems'

const Menu = ({route = window.location.pathname}) => {
    const isNotSelectedMenu = item => (item.link !== route)

    const markItemAsActive = item => 
        isNotSelectedMenu(item) 
            ? item
            : { ...item, isActive: true }

    const menuItemsActivated = menuItems.map(markItemAsActive)

    return (
        <div className="tabs is-centered">
            <ul>
                { 
                    menuItemsActivated.map((item, i) => <MenuItem key={i} item={item} /> )
                }
            </ul>
        </div>
    )
}

export default Menu
