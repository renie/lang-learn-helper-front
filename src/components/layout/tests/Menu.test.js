import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'

import Menu from '../Menu'

it('renders correctly no option', () => {
    const tree = renderer
        .create(<Router><Menu /></Router>)
        .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly no option', () => {
    const tree = renderer
        .create(<Router><Menu route='/editor' /></Router>)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
