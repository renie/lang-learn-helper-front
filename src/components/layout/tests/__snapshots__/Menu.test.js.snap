// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly no option 1`] = `
<div
  className="tabs is-centered"
>
  <ul>
    <li
      className="is-active"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/editor"
        onClick={[Function]}
      >
        Write Essay
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/word/new"
        onClick={[Function]}
      >
        New Word
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders correctly no option 2`] = `
<div
  className="tabs is-centered"
>
  <ul>
    <li
      className=""
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        href="/editor"
        onClick={[Function]}
      >
        Write Essay
      </a>
    </li>
    <li
      className=""
    >
      <a
        href="/word/new"
        onClick={[Function]}
      >
        New Word
      </a>
    </li>
  </ul>
</div>
`;
